### PR TITLE
Add in-app alert to let user know if a new version is available.

### PR DESCRIPTION
**Pull request by bmarty**
Add in app new version alert on nightly build version of the app, following this recipe: https://firebase.google.com/docs/app-distribution/set-up-alerts?platform=android, step 1, step 2 and step 4.

Screenshot 2022-07-22 at 15 33 51

Still not working on my device with my account.

I have got this error when sign in to my Google Account:

image

The step 4 suggest that the app have to be installed from Firebase to test the implementation. Let's merge this PR, wait for the next nightly build and get the app from Firebase to see if it's working?

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -15,6 +15,7 @@ def gradle = "7.1.3"
 def kotlin = "1.6.21"
 def kotlinCoroutines = "1.6.4"
 def dagger = "2.42"
+def appDistribution = "16.0.0-beta03"
```
Comment (ganfra) [inline dependencies.gradle:18]: Is there any stable version instead?

File: vector/src/fdroid/java/im/vector/app/nightly/NightlyProxy.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.nightly
+
+import javax.inject.Inject
+
+class NightlyProxy @Inject constructor() {
+    fun updateIfNewReleaseAvailable() {
```
Comment (ganfra) [inline vector/src/fdroid/java/im/vector/app/nightly/NightlyProxy.kt:22]: could be `fun updateIfNewReleaseAvailable()  = Unit`

**Review by ganfra — APPROVED** [review]
Small remarks

File: dependencies.gradle (hunk 1 above)
Comment (bmarty) [reply dependencies.gradle:18]: I was just following the doc: https://firebase.google.com/docs/app-distribution/set-up-alerts?platform=android

image

I guess a stable version will be release someday and Dependabot will do the rest.

File: vector/src/fdroid/java/im/vector/app/nightly/NightlyProxy.kt (hunk 2 above)
Comment (bmarty) [reply vector/src/fdroid/java/im/vector/app/nightly/NightlyProxy.kt]: https://github.com/vector-im/element-android/pull/6629/commits/4dd5fa7051b373d1a622c6158723fd370d0f8749

**Comment by bmarty** [comment]
Adding a post merging commit cc0c9c262cc128bf00028693068a25153ef4fef4 directly on develop.

**Comment by witchent** [comment]
I always get this message as soon as I enter the timeline - be it from a chat or from closing the app, which is super annoying. Could you add a third button "don't bother me again" or something next to "not now" and "turn on"?

**Pull request merged**